File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
       int ct_z=0;
       for(auto v:nums){
        if(v==0) ct_z++;
       }
         vector<int>ans(nums.size(),0);

       if(ct_z>=2){
         return ans;
       }
       if(ct_z==1){
         int prod=1;
         for(int i=0;i<nums.size();i++){
            if(nums[i]!=0) {
                prod*=nums[i];
                ans[i]=0;
            }

         }
         for(int i=0;i<nums.size();i++){
            if(nums[i]==0) ans[i]=prod;
         }
         return ans;

       }
        int prod = 1;
       for(int i=0;i<nums.size();i++){
        prod*=nums[i];

       }
       for(int i=0;i<nums.size();i++){
        ans[i]=prod/nums[i];
       }
       return ans;

    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        final_ans=[]
        def ps(nums,i,ans):
            if i==len(nums):
                final_ans.append(ans.copy())
                return
            ans.append(nums[i])
            ps(nums,i+1,ans)
            ans.pop()
            j=i
            while j<len(nums) and nums[j]==nums[i]:
                j+=1
            ps(nums,j,ans)
        ps(nums,0,[])
        return final_ans